src/CheckVariables.ts

```diff
@@ -231,6 +231,13 @@ export const checkVariablesOnLoad = (data: PlayerSave) => {
         player.shopUpgrades.tesseractToQuark = 0;
         player.shopUpgrades.hypercubeToQuark = 0;
     }
+    for (const i in shopData) { //Since you cant increase SingularityCount without calling it
+        if (shopData[i as keyof typeof shopData].tier === 'Reincarnation') {
+            shopData[i as keyof typeof shopData].refundable = player.singularityCount >= 20 ? false : true;
```

Comment:
Would this make all shop items refundable below 20? Including instantChallenge, and shopTalisman, (and potions). I think you mean to only change the ones where refundable === true.

I like it because it makes hide maxed work better. I think it belongs in the updateSingularityMilestoneAwards in Reset.ts, though.

Reply:
Oh wait that probably true, will deal with it shortly
About putting it in reset, don't really want because it has to happen on refresh and preferably only once per reload
It can be moved anywhere into reload function, but I don't know how often it is called and when